app/src/views/features/sessions/SessionViewer/SavedSessionViewer.tsx

```diff
@@ -125,12 +124,17 @@ const SavedSessionViewer: React.FC = () => {
   );
 
   useEffect(() => {
+    if (isInsideIframe) {
+      trackSavedSessionViewed("embed");
+      return;
+    }
+
     if ((location.state as NavigationState)?.fromApp) {
-      trackSavedSessionViewedFromApp();
+      trackSavedSessionViewed("app");
     } else {
-      trackSavedSessionViewedFromLink();
+      trackSavedSessionViewed("link");
     }
-  }, [id, location.state]);
+  }, [id, location.state, isInsideIframe]);
```

Comment:
if possible, this should be handled where the page view events are sent, so in the app router